Durch Gson serializierbareren DateTimeFormatter erstellen

Aktuell ist es nicht möglich das zu einem User die offerListe gespeichert wird, da der DateTimeFormatter nicht serializiert werden kann:

https://stackoverflow.com/questions/39192945/serialize-java-8-localdate-as-yyyy-mm-dd-with-gson

` gson.toJson(user);`

FIXME Caused by: java.lang.IllegalArgumentException: class java.text.DecimalFormat declares multiple JSON fields named maximumFractionDigits

at com.github.neighbortrader.foodboardapp.clientmodel.User.saveToSharedPreferences(User.java:90)

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -33,7 +33,7 @@ android {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
 
-            resValue "string", "BASE_URL", "http://192.168.0.4:3000/api/"
+            resValue "string", "BASE_URL", "192.168.178.27:3000/api/"
             resValue "string", "GET_ALL_OFFERS_URL", "Offer"
             resValue "string", "CREATE_NEW_USER", "User"
             resValue "string", "CREATE_NEW_OFFER", "Offer"
@@ -44,7 +44,7 @@ android {
         debug {
             signingConfig debug.signingConfig
 
-            resValue "string", "BASE_URL", "http://192.168.0.4:3000/api/"
+            resValue "string", "BASE_URL", "http://192.168.178.27:3000/api/"
             resValue "string", "GET_ALL_OFFERS_URL", "Offer"
             resValue "string", "CREATE_NEW_USER", "User"
             resValue "string", "CREATE_NEW_OFFER", "Offer"

--- a/app/src/main/java/com/github/neighbortrader/foodboardapp/handler/clientmodelHandler/UserHandler.java
+++ b/app/src/main/java/com/github/neighbortrader/foodboardapp/handler/clientmodelHandler/UserHandler.java
@@ -35,7 +35,7 @@ public class UserHandler {
     public static User generateRandomUser() {
         String username = UUID.randomUUID().toString();
         String password = "adm!n9Asswor!d";
-        String email = "test@testemail.com";
+        String email = String.format("%s@testmail.com", username);
         Address address = new Address("Teststareße", "12a", "10315", "Berlin");
 
         return User.userBuilder(username, password, email, address);
@@ -47,7 +47,7 @@ public class UserHandler {
         if (user != null) {
             Gson gson = GsonHandler.getGsonInstance();
 
-            String userToSaveAsJsonString = null; //gson.toJson(user);
+            String userToSaveAsJsonString = gson.toJson(user);
 
             SharedPreferences sharedPreferences = ContextHandler.getAppContext().getSharedPreferences(SHARED_PREFERENCES_FILE_USER_INFO, MODE_PRIVATE);
             SharedPreferences.Editor editor = sharedPreferences.edit();

